packages/@tinacms/fields/src/Wysiwyg/state/plugins/Menu/Menu.tsx

```diff
@@ -85,6 +86,37 @@ const LinkControl = markControl({
 
 export const Menu = (props: Props) => {
   const { view, bottom = false, frame, theme } = props
+  const [menuFixed, setMenuFixed] = useState(false)
+  const [menuOffset, setMenuOffset] = useState(0)
+  const [menuWidth, setMenuWidth] = useState()
+  const menuRef: any = useRef()
+
+  const handleScroll = () => {
+    // Need to know the Y coord of the bottom of the div that contains the text
+    const textAreaBottom =
+      menuRef.current.parentElement.nextSibling.offsetHeight + menuOffset
+
+    if (
+      window.scrollY > menuOffset &&
+      window.scrollY < textAreaBottom &&
+      !menuFixed
+    ) {
+      // Need to remember the menu original position and width
+      setMenuOffset(menuRef.current.offsetTop)
+      setMenuWidth(menuRef.current.offsetWidth)
```

Comment:
I got `cannot read offsetWidth of null` during testing at one point; I added `menuRef.current` to the conditional and that seemed to fix it.